src/pumipic_adjacency.tpp

```diff
@@ -369,47 +376,54 @@ namespace pumipic {
                                 o::Write<o::LO> lastExit, o::Bytes side_is_exposed,
                                 bool requireIntersection,
                                 o::Write<o::LO> xFace) {
-    auto checkExposedEdges = PS_LAMBDA(const int e, const int pid, const int mask) {
-      if( mask > 0 && !ptcl_done[pid] ) {
-        assert(lastExit[pid] != -1);
-        const o::LO bridge = lastExit[pid];
-        const bool exposed = side_is_exposed[bridge];
-        ptcl_done[pid] = exposed;
-        if (exposed && requireIntersection) {
-          xFace[pid] = lastExit[pid];
-        }
-        else {
-          elem_ids[pid] = exposed ? -1 : elem_ids[pid]; //leaves domain if exposed
+    auto checkIntersectionwGeomBoundary = PS_LAMBDA(const int e, const int pid, const int mask) {
```

Comment:
Rename this to `intersectsWithGeometryInterface` when you write functions that will return a bool (or array of bool) put a question mark after the name in your head. If the answer to that question matches the result of your boolean then it will be clear what the function is doing.

With the current name, if you say "check the intersection with the geometry boundary?` the name does not give any clues about what true means and what false means.